### PR TITLE
[move] Add example of user code organized into multiple packages

### PR DESCRIPTION
**Pull request by awelc**
This resolves https://github.com/MystenLabs/sui/issues/5517 by providing an example on how to structure user code with multiple packages.

### CODE REVIEW

**Comment by amnn** [comment]
The example looks good!  We should likely document it [here](https://examples.sui.io/basics/move-toml.html) as well.  I'll leave it to @randall-Mysten to give the final stamp from a docs perspective.

File: sui_programmability/examples/multi_package/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+An example of how to structure your code to include another user-level package as a dependency.
+
+The main things to pay attention to is to make sure that the dependency name (in this example
+DepPackage in main_package/Move.toml file's `[dependency]` section) is the same as the name of the
```
Comment (randall-Mysten) [inline sui_programmability/examples/multi_package/README.md:4]: is the same as the name of the
package (in dep_package/Move.toml file's `[package]` section) are exactly the same

seems redundant - make sure that the dep name is the same as the name of the package (...) are exactly the same...

**Review by amnn — COMMENTED** [review]
*sigh* I left a bunch of comments on here this morning, and then forgot that they were pending.

File: sui_programmability/examples/multi_package/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+An example of how to structure your code to include another user-level package as a dependency.
+
+The main things to pay attention to is to make sure that the dependency name (in this example
```
[the remark this answers is not in the document]
Comment (randall-Mysten) [reply sui_programmability/examples/multi_package/README.md:3]: it sounds like there is only 1 thing to pay attention to?
What if it said "Make sure that the dependency name in this example"

File: sui_programmability/examples/multi_package/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (randall-Mysten) [reply sui_programmability/examples/multi_package/README.md:4]: why is this in brackets?

[the remark this answers is not in the document]
Comment (amnn) [reply sui_programmability/examples/multi_package/README.md:4]: It is referring to the syntax in TOML files where table (key-value section) headers are denoted by square brackets.

**Comment by awelc** [comment]
Thank you for the comments and suggestions @amnn and @randall-Mysten. And sorry for the sloppy writing (among other things, started thinking that there would be more things to pay attention to and this never materialized but artifacts remained...)

**Comment by randall-Mysten** [comment]
> Thank you for the comments and suggestions @amnn and @randall-Mysten. And sorry for the sloppy writing (among other things, started thinking that there would be more things to pay attention to and this never materialized but artifacts remained...)

Thanks for updating the docs. Happy to partner on this earlier in the process so we can iterate and fine-tune before creating the PR.

**Review by randall-Mysten — APPROVED** [review]

**Pull request merged**